rasterio/crs.pyx

```diff
@@ -284,14 +287,18 @@ cdef class CRS:
         CRSError
 
         """
+        if self._geodetic_crs is not None:
+            return self._geodetic_crs if self._geodetic_crs is not False else None
         cdef CRS obj = CRS.__new__(CRS)
```

Comment:
This will need to go on top of the function in the approach I am suggesting.

Reply:
did another comment get left out, I am a little confused what needs to me moved to the top of the function.

Reply:
Sorry! I fumbled that one. Let's move `cdef CRS obj = CRS.__new__(CRS)` to the head of the function code. We can't `cdef` within `if/else` but we can use cdef'd variables within.